arcgis-ios-sdk-samples/Utility Network/Display utility association/DisplayUtilityAssociationVC.swift

```diff
@@ -0,0 +1,141 @@
+        utilityNetwork.load { [weak self] error in
+            if let error = error {
+                self?.presentAlert(error: error)
+                return
+            } else {
+                guard let self = self else { return }
+                // Get all the edges and junctions in the network.
+                let edges = self.utilityNetwork.definition.networkSources.filter { $0.sourceType == .edge }
+                let junctions = self.utilityNetwork.definition.networkSources.filter { $0.sourceType == .junction }
+
+                // Add all edges that are not subnet lines to the map.
+                edges.filter { $0.sourceUsageType != .subnetLine }.forEach { source in
+                    self.mapView.map?.operationalLayers.add(AGSFeatureLayer(featureTable: source.featureTable))
+                }
+                // Add all the junctions to the map.
+                junctions.forEach { source in
+                    self.mapView.map?.operationalLayers.add(AGSFeatureLayer(featureTable: source.featureTable))
+                }
+                
+                // Create a renderer for the associations.
+                let attachmentValue = AGSUniqueValue(description: "Attachment", label: "", symbol: self.attachmentSymbol, values: [AGSUtilityAssociationType.attachment])
+                let connectivityValue = AGSUniqueValue(description: "Connectivity", label: "", symbol: self.connectivitySymbol, values: [AGSUtilityAssociationType.connectivity])
+                self.associationsOverlay.renderer = AGSUniqueValueRenderer(fieldNames: ["AssociationType"], uniqueValues: [attachmentValue, connectivityValue], defaultLabel: "", defaultSymbol: nil)
+                
+                // Populate the legened.
+                self.attachmentSymbol.createSwatch(withBackgroundColor: nil, screen: .main) { [weak self] (image, error) in
+                    if let error = error {
+                        print("Error creating swatch: \(error)")
+                    } else {
+                        self?.attachmentBBI.image = image?.withRenderingMode(.alwaysOriginal)
+                    }
+                }
+                self.connectivitySymbol.createSwatch(withBackgroundColor: nil, screen: .main) { [weak self] (image, error) in
+                    if let error = error {
+                        print("Error creating swatch: \(error)")
+                    } else {
+                        self?.connectivityBBI.image = image?.withRenderingMode(.alwaysOriginal)
+                    }
+                }
+                self.addAssociationGraphics()
+                self.mapViewDidChange()
+            }
+        }
+    }
+    
+    func addAssociationGraphics() {
+        // Check if the current viewpoint is outside of the max scale.
+        if let targetScale = mapView.currentViewpoint(with: .centerAndScale)?.targetScale {
+            if targetScale >= maxScale {
+                return
+            }
+        }
```

Comment:
Since the optional is the viewpoint, not the target scale, I recommend making that the variable. Also, the two `if` statements could be combined:

```suggestion
        if let viewpoint = mapView.currentViewpoint(with: .centerAndScale),
            viewpoint.targetScale >= maxScale {
            return
        }
```